src/Analyzers/CSharp/CodeFixes/UseInterpolatedString/CSharpUseInterpolatedStringCodeFixProvider.cs

```diff
@@ -0,0 +1,103 @@
+            return;
+
+        var newNode = CreateInterpolatedString(node);
+        editor.ReplaceNode(node, newNode);
+    }
+
+    private static InterpolatedStringExpressionSyntax CreateInterpolatedString(SyntaxNode stringLiteralExpressionNode)
+    {
+        var token = stringLiteralExpressionNode.GetFirstToken();
+        var originalTokenKind = token.Kind();
+        var isVerbatimString = token.IsVerbatimStringLiteral();
+        var newTokenKind = GetKind(originalTokenKind);
+        var startTokenKind = GetStartToken(originalTokenKind, isVerbatimString);
+        var text = GetTextWithoutQuotes(token, isVerbatimString);
+        var endTokenKind = GetEndToken(originalTokenKind);
+
+        var newNode = SyntaxFactory.InterpolatedStringText(
+            SyntaxFactory.Token(
+                leading: SyntaxTriviaList.Empty,
+                kind: newTokenKind,
+                text: text,
+                valueText: text,
+                trailing: SyntaxTriviaList.Empty))
+            .WithTriviaFrom(stringLiteralExpressionNode);
+
+        return SyntaxFactory.InterpolatedStringExpression(
+            stringStartToken: SyntaxFactory.Token(startTokenKind),
+            contents: [newNode],
+            stringEndToken: SyntaxFactory.Token(endTokenKind));
+    }
+
+    private static SyntaxKind GetKind(SyntaxKind kind) => kind switch
+    {
+        SyntaxKind.SingleLineRawStringLiteralToken => SyntaxKind.InterpolatedStringTextToken,
+        SyntaxKind.MultiLineRawStringLiteralToken => SyntaxKind.InterpolatedStringTextToken,
+        _ => SyntaxKind.InterpolatedStringTextToken,
+    };
+
+    private static SyntaxKind GetStartToken(SyntaxKind kind, bool isVerbatimString) => kind switch
+    {
+        SyntaxKind.SingleLineRawStringLiteralToken => SyntaxKind.InterpolatedSingleLineRawStringStartToken,
+        SyntaxKind.MultiLineRawStringLiteralToken => SyntaxKind.InterpolatedMultiLineRawStringStartToken,
+        _ => isVerbatimString ? SyntaxKind.InterpolatedVerbatimStringStartToken : SyntaxKind.InterpolatedStringStartToken,
+    };
+
+    private static string GetTextWithoutQuotes(SyntaxToken token, bool isVerbatimString)
+    {
+        var text = token.Text.Replace("{", "{{").Replace("}", "}}");
+        var kind = token.Kind();
+        var startIndex = GetStartIndex(kind, isVerbatimString);
+        var endIndex = GetEndIndex(kind);
+        return text[startIndex..^endIndex];
+    }
+
+    private static SyntaxKind GetEndToken(SyntaxKind kind) => kind switch
+    {
+        SyntaxKind.SingleLineRawStringLiteralToken => SyntaxKind.InterpolatedRawStringEndToken,
+        SyntaxKind.MultiLineRawStringLiteralToken => SyntaxKind.InterpolatedRawStringEndToken,
+        _ => SyntaxKind.InterpolatedStringEndToken,
+    };
+
+    private static int GetStartIndex(SyntaxKind kind, bool isVerbatimString) => (isVerbatimString ? 1 : 0)
+        + kind switch
```

Comment:
I find these 2 lines hard to read. Given that this is not a trivial switch I would recommend using block body here and _not_ put operands of an addition on different lines. Maybe for clarify even use separate local for storing a result of the switch and the return `(isVerbatimString ? 1 : 0) + newLocal`